io/plugins/eu.esdihumboldt.hale.io.xls.ui/src/eu/esdihumboldt/hale/io/xls/ui/XLSInstanceExportConfigurationPage.java

```diff
@@ -138,25 +140,62 @@ public String getText(Object element) {
 			});
 			featureTypeTable.setContentProvider(ArrayContentProvider.getInstance());
 
-			Collection<? extends TypeDefinition> relevantTypes = getWizard().getProvider()
-					.getTargetSchema().getMappingRelevantTypes();
-
-			ArrayList<QName> tableContent = new ArrayList<>();
-			for (TypeDefinition typeDefinition : relevantTypes) {
-				tableContent.add(typeDefinition.getName());
-			}
+			featureTypeTable.setInput(
+					getWizard().getProvider().getTargetSchema().getMappingRelevantTypes());
 
-			featureTypeTable.setInput(relevantTypes);
 			featureTypeTable.addCheckStateListener(new ICheckStateListener() {
 
 				@Override
 				public void checkStateChanged(CheckStateChangedEvent event) {
+					// Programmatic action to toggle the state
+					selectAll.setSelection(
+							featureTypeTable.getCheckedElements().length == featureTypeTable
+									.getTable().getItemCount());
+
 					page.layout();
 					page.pack();
 					setPageComplete(validate());
 				}
 			});
 
+			featureTypeTable.setCheckStateProvider(new ICheckStateProvider() {
+
+				@Override
+				public boolean isChecked(Object element) {
+					if (!(element instanceof TypeDefinition))
+						return false;
+					return checkboxState(element);
+				}
+
+				@Override
+				public boolean isGrayed(Object element) {
+					if (!(element instanceof TypeDefinition))
+						return false;
+					return checkboxState(element);
+				}
+
+				/**
+				 * @param element
+				 * @return true if the button cannot be selected
+				 */
+				private boolean checkboxState(Object element) {
+					InstanceService ins = PlatformUI.getWorkbench()
+							.getService(InstanceService.class);
+					// select all source type which has at least one instance
+					if (!ins.getInstances(DataSet.SOURCE)
```

Comment:
Instead of getting the data I think it would be more efficient to get the information from the `InstanceService` for which types there is data.
See https://github.com/halestudio/hale/blob/019cb4e73e00d1b022b11c7a54177db69d903931/ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/instance/InstanceService.java#L57

Reply:
Than maybe the same approach should be used here
https://github.com/halestudio/hale/blob/019cb4e73e00d1b022b11c7a54177db69d903931/io/plugins/eu.esdihumboldt.hale.io.csv.ui/src/eu/esdihumboldt/hale/io/csv/ui/InstanceExportConfigurationPage.java#L54
as well.

Reply:
Yes, I agree.